controllers/storagecluster/provider_server.go

```diff
@@ -454,6 +454,7 @@ func getOnboardingJobObject(instance *ocsv1.StorageCluster) *batchv1.Job {
 			Namespace: instance.Namespace,
 		},
 		Spec: batchv1.JobSpec{
+			TTLSecondsAfterFinished: util.ToPointer[int32](0), //Eligible to delete automatically when job completes
```

Comment:
- move comment to line above and not inline
- TTL works for both Job Complete & Failed states, either change the comment to reflect that or just use `job finishes`